### PR TITLE
update codegens to move the last instance instead of clone

### PR DESCRIPTION
**Pull request by abbychau**
#304

### CODE REVIEW

File: compiler/src/codegen.rs [hunk 1]
```diff
@@ -567,4 +571,4 @@ pub fn gen(
 
 pub fn protoc_gen_grpc_rust_main() {
     compiler_plugin::plugin_main(gen);
-}
+}
```
Comment (ice1000) [inline compiler/src/codegen.rs]: Please don't remove this line end.

File: compiler/src/prost_codegen.rs [hunk 2]
```diff
@@ -535,4 +544,4 @@ fn generate_method_bind(service_name: &str, method: &Method, buf: &mut String) {
     buf.push_str(", move |ctx, req, resp| instance.");
     buf.push_str(&method.name);
     buf.push_str("(ctx, req, resp));\n");
-}
+}
```
Comment (ice1000) [inline compiler/src/prost_codegen.rs]: Ditto

**Comment by ice1000** [comment]
It is important for you to pass CI tests. It's not possible to merge if CI is not happy (we do fmt check on CI).

File: compiler/src/codegen.rs (hunk 1 above)
Comment (abbychau) [reply compiler/src/codegen.rs]: Done

File: compiler/src/prost_codegen.rs (hunk 2 above)
Comment (abbychau) [reply compiler/src/prost_codegen.rs]: Done

**Pull request merged**